### ISSUE TITLE
What is polymer's approach to performant DOM?

### ISSUE THREAD

**Issue by devinivy** [richardanaya/webblock#1]
In your readme it says,

> Polymer... tries to re-implement it's own way of "how do I update my dom nodes quickly" (virtual dom) when other more popular and suitable ways exist

Could you summarize the approach Polymer takes to performant DOM?  I always thought it was simply that it performs minimal changes to the actual DOM by getting the programmer to describe specific changes to component properties (using `el.set()`, `el.splice()`, `el.push()`).  This isn't really akin to virtual DOM from my point of view, so I imagine there's something else at play that I hadn't picked-up on.

**Comment by richardanaya** [comment]
To be honest, I don't know the deep details.  They have some observer system that underlies everything and connects observable attributes to parts of the dom.
